addons/mail/models/discuss/discuss_channel.py

```diff
@@ -59,6 +59,9 @@ def _generate_random_token(self):
         compute='_compute_channel_partner_ids', inverse='_inverse_channel_partner_ids',
         search='_search_channel_partner_ids')
     channel_member_ids = fields.One2many('discuss.channel.member', 'channel_id', string='Members')
+    owner_id = fields.Many2one("discuss.channel", help="Parent channel", ondelete="cascade")
```

Comment:
I don't get why we can't use the generic `parent_id` here, it's a `discuss.channel` linking to a `discuss.channel`, and this field doesn't exist.
There is no need for a fancy name, `parent_id` should be the go to.

Reply:
Agreed for `parent_id`, I updated the code.

Reply:
`parent_id` is fine, but could pose confusion and also might compromise future improvements.

- confusion from message.parent_id, which is used to refer to message being replied to in discuss.channel.
- if we make mail.message also inherit from mail.thread (like the message behaves like a thread, a bit like how Twitter does it), then there's conflict between `DiscussChannel/MailThread.parent_id` and `MailMessage.parent_id`. This would necessarily require costly migration scripts in the case the rename happens on `MailMessage.parent_id`.

At least `parent_id` is better than `owner_id` IMO.